cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -197,6 +197,31 @@ geometry::TriangleMesh TriangleMesh::FromLegacy(
                 core::eigen_converter::EigenVector3dVectorToTensor(
                         mesh_legacy.triangle_normals_, float_dtype, device));
     }
+    if (mesh_legacy.HasTriangleUvs()) {
+        // Legacy uses UVs as triangle attributes
+        // Check length and issue warning on mismatch
+        if (mesh_legacy.triangle_uvs_.size() != mesh_legacy.vertices_.size()) {
```

Comment:
UVs map vertices of each triangle to a 2D coordinate [Image of (h, w) : (1, 1)]. 
So, each vertex in a triangle has a 2D UV coordinate, therefore 3 such coordinates per triangle.

A vertex can have multiple UV coordinates (for example, if we un-wrap a cylinder on a 2D plane, the cut vertices will be having 2 UV coordinates).

My suggestion would be, to use `texture_uvs`, as a `triangle tensor map attribute`, of shape `[N, 3, 2]`, where `N` is the Length of `Triangle Indices`. 
In Legacy, we have a vector of `Eigen::Vector2d` of length `3xN`,
so, we need to reshape [N x 3, 2] (output of `EigenVector2dVectorToTensor`) -> [N, 3, 2] in tensor.

so the following might work: (not tested)
```cpp
// we have a function in legacy to check UV shape.
if(!mesh.HasTriangleUvs()) {
            utility::LogWarning(
                    "Ignoring UV coordinates since length(UV) ({}) != "
                    "3 x n_triangles ({})",
                    mesh_legacy.triangle_uvs_.size(),
                    mesh_legacy.triangles_.size());
} else {
            mesh.SetVertexAttr(
                    "texture_uvs",
                    core::eigen_converter::EigenVector2dVectorToTensor(
                            mesh_legacy.triangle_uvs_, float_dtype, device)).Reshape(mesh_legacy.triangles_.size(), 3, 2);
}
```

For `Tensor` -> `Legacy`
We will need to `Reshape` `[N, 3, 2]` back to `[N x 3, 2]`, and then use `TensorToEigenVector2dVector`